src/main/java/org/opensearch/ad/rest/RestCreateDetectorAction.java

```diff
@@ -1,134 +1,241 @@
+import org.opensearch.client.opensearch.core.IndexRequest;
+import org.opensearch.client.opensearch.core.IndexResponse;
+import org.opensearch.client.opensearch.indices.CreateIndexRequest;
+import org.opensearch.client.opensearch.indices.CreateIndexResponse;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
 import org.opensearch.common.xcontent.XContentType;
 import org.opensearch.extensions.rest.ExtensionRestRequest;
 import org.opensearch.extensions.rest.ExtensionRestResponse;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.NestedQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.plugins.SearchPlugin;
 import org.opensearch.rest.RestHandler.Route;
 import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.rest.RestStatus;
 import org.opensearch.sdk.ExtensionRestHandler;
+import org.opensearch.search.aggregations.BaseAggregationBuilder;
+import org.opensearch.search.aggregations.bucket.filter.FilterAggregationBuilder;
+import org.opensearch.search.aggregations.bucket.filter.InternalFilter;
+import org.opensearch.search.aggregations.metrics.InternalSum;
+import org.opensearch.search.aggregations.metrics.SumAggregationBuilder;
+
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import jakarta.json.stream.JsonParser;
 
 public class RestCreateDetectorAction implements ExtensionRestHandler {
+    private final Logger logger = LogManager.getLogger(RestCreateDetectorAction.class);
+    private AnomalyDetectorExtension anomalyDetectorExtension = new AnomalyDetectorExtension();
+    private OpenSearchClient sdkClient = anomalyDetectorExtension.getClient();
 
-    private static final String GREETING = "Hello, %s!";
-    private String worldName = "World";
-    private List<String> worldAdjectives = new ArrayList<>();
-    private Random rand = new Random();
+    public RestCreateDetectorAction() throws IOException {}
 
     @Override
     public List<Route> routes() {
-        return List.of(new Route(GET, "/hello"), new Route(PUT, "/hello/{name}"));
+        return List.of(new Route(POST, "/detectors"));
     }
 
-    @Override
-    public ExtensionRestResponse handleRequest(ExtensionRestRequest request) {
-        Method method = request.method();
+    private String getAnomalyDetectorMappings() throws IOException {
+        URL url = AnomalyDetectionIndices.class.getClassLoader().getResource(ANOMALY_DETECTORS_INDEX_MAPPING_FILE);
+        return Resources.toString(url, Charsets.UTF_8);
+    }
+
+    private IndexResponse indexAnomalyDetector(AnomalyDetector anomalyDetector) throws IOException {
+        AnomalyDetector detector = new AnomalyDetector(
```

Comment:
It looks here like we're pretty much cloning the input parameter.  Why is this needed?

Reply:
IndexRequest requires a model class object to document it with the index.
```
IndexRequest<AnomalyDetector> indexRequest = new IndexRequest.Builder<AnomalyDetector>()
            .index(ANOMALY_DETECTORS_INDEX)
            .id("1")
            .document(detector)
            .build();